src/OpenTelemetry/Metrics/Export/PushMetricController.cs

```diff
@@ -0,0 +1,122 @@
+﻿// <copyright file="PushMetricController.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Metrics.Configuration;
+using static OpenTelemetry.Metrics.Configuration.MeterFactory;
+
+namespace OpenTelemetry.Metrics.Export
+{
+    internal class PushMetricController
+    {
+        private readonly TimeSpan pushInterval;
+        private readonly Task worker;
+        private MetricExporter metricExporter;
+        private MetricProcessor metricProcessor;
+        private Dictionary<MeterRegistryKey, Meter> meters;
+
+        public PushMetricController(Dictionary<MeterRegistryKey, Meter> meters,
+            MetricProcessor metricProcessor,
+            MetricExporter metricExporter,
+            TimeSpan pushInterval,
+            CancellationTokenSource cts)
+        {
+            this.meters = meters;
+            this.metricProcessor = metricProcessor;
+            this.metricExporter = metricExporter;
+            this.pushInterval = pushInterval;
+            this.worker = Task.Factory.StartNew(
+                s => this.Worker((CancellationToken)s), cts.Token);
+        }
+
+        private async Task Worker(CancellationToken cancellationToken)
+        {
+            try
+            {
+                List<Metric<long>> longMetricToExport = new List<Metric<long>>();
+                List<Metric<double>> doubleMetricToExport = new List<Metric<double>>();
+
+                await Task.Delay(this.pushInterval, cancellationToken).ConfigureAwait(false);
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    var sw = Stopwatch.StartNew();
+
+                    foreach (var meter in this.meters.Values)
+                    {
+                        (meter as MeterSdk).Collect();
```

Comment:
is this cast necessary? Can we store metrics as MeterSdk if we are sure it would be the type for sure